src/coreclr/jit/gentree.cpp

```diff
@@ -12913,17 +12913,19 @@ GenTree* Compiler::gtFoldTypeCompare(GenTree* tree)
 
     // Screen for the right kinds of operands
     GenTree* const         op1     = tree->AsOp()->gtOp1;
-    const TypeProducerKind op1Kind = gtGetTypeProducerKind(op1);
-    if (op1Kind == TPK_Unknown)
-    {
-        return tree;
-    }
-
     GenTree* const         op2     = tree->AsOp()->gtOp2;
+    const TypeProducerKind op1Kind = gtGetTypeProducerKind(op1);
     const TypeProducerKind op2Kind = gtGetTypeProducerKind(op2);
-    if (op2Kind == TPK_Unknown)
+
+    // Fold "typeof(handle) cmp null"
+    if ((op2->IsIntegralConst(0) && (op1Kind == TPK_Handle)) || (op1->IsIntegralConst(0) && (op2Kind == TPK_Handle)))
```

Comment:
```suggestion
    if (((op2Kind == TPK_Null) && (op1Kind == TPK_Handle)) || ((op1Kind == TPK_ Null) && (op2Kind == TPK_Handle)))
```

Reply:
Doesn't `gtGetTypeProducerKind` already look for nulls?

Reply:
Ah, indeed, fixed